extensions/resource-deployment/src/ui/validation/validations.ts

```diff
@@ -0,0 +1,216 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { EOL } from 'os';
+import { throwUnless } from '../../common/utils';
+import * as loc from '../../localizedConstants';
+import { InputComponent } from '../modelViewUtils';
+import { RadioGroupLoadingComponentBuilder } from '../radioGroupLoadingComponentBuilder';
+
+export interface ValidationResult {
+	valid: boolean;
+	message?: string;
+}
+
+export type Validator = () => Promise<ValidationResult>;
```

Comment:
is it really needed to introduce this validator type? this could be simplified to a validate method

Reply:
Backward compatibility. This type exists in a lot of our codebase, but I can investigate removing it. I agree this is not useful.

Reply:
Ok keeping Validator type but simplifying code so that validate can be called simply instead of having to use ()().

Reply:
Kindly note that Validator defined in modelviewutils will be replaced by the definition here. This allows us to seamlessly slide in the new validator class for existing usages.